Add cron job for link checker

brenetic opened the pull request:
- Added Redis for doing distributed lock so that the rake task only runs on one machine.
- Added Whenever gem to schedule cron job to run the link checker task daily.
- Added notify_passive_check script to notify Icinga of success/failure.
- Moved all other import tasks to whenever (see: alphagov/govuk-puppet#4591)

[Trello card](https://trello.com/c/KBT9RC9q/406-check-the-status-code-of-links)

[Related PR](https://github.com/alphagov/govuk-puppet/pull/4587)

brenetic committed:
Add redis-namespace and redis-lock gems

brenetic committed:
Add redis configuration

Redis is required for distrubuted lock during the running of cronjob
rake tasks. This is copied from Transition which is already doing this.

brenetic committed:
Add Whenever gem and link checker scheduled job

Wrapped link checker rake task in distributed lock and scheduled it to
run daily at 2am

brenetic committed:
Add Icinga passive check

We changed the distributed_lock to take callbacks for what to do when
the lock is obtained and when it is not obtained. This follows the
pattern in the Publisher script where if the lock is not obtained we
tell Icinga it was a success but we did not actually run the task.

We now communicate with Icinga using a new notify_passive_check script
which was added to Puppet.

alexmuller commented on lib/services.rb:
```diff
@@ -1,10 +1,27 @@
+  end
+
+  def self.icinga_check(service_desc, code, message)
+    `notify_passive_check "#{service_desc}" #{code} "#{message}"`
```
You might want to use the full path here, I'm not sure what the deploy user's `$PATH` looks like.

brenetic commented:
The notification to Icinga will currently run in dev whenever the rake task is run. Do we not want to do that?

brenetic committed:
Add links and service interaction imports to cron

These used to be controlled by jenkins jobs, but we want to keep
everything consistent so we've moved them into the crontab controlled by
the app.  We also change the links import to run at 1am instead of 5am
so that it runs before the link checker at 2am.

brenetic commented:
Having tested running the rake tasks in development, we definitely want to not talk to icinga in dev as it breaks.  Anyone got any betters ideas than adding a `unless Rails.env.development?` guard clause to the `Service.icinga_check`?

sihugh commented:
I think the guard clause is fine, and this stuff is really useful.
